jetty-websocket/websocket-core-server/src/main/java/org/eclipse/jetty/websocket/core/server/ServerUpgradeRequest.java

```diff
@@ -51,20 +52,17 @@
     public ServerUpgradeRequest(WebSocketNegotiation negotiation) throws BadMessageException
     {
         this.negotiation = negotiation;
-        HttpServletRequest httpRequest = negotiation.getRequest();
-        this.queryString = httpRequest.getQueryString();
-        this.secure = httpRequest.isSecure();
+        this.request = negotiation.getRequest();
```

Comment:
OK I'm now totally confused by this class?
I thought it had a reference to the request that at some point was nulled out and all the required request fields copied.  But this class is holding it for all time and calling it at any time it likes?

How is this different to `UpgradeHttpServletRequest`?

Can you explain what is going on?  Perhaps in the javadoc for the class?

Reply:
This is just the interface used during the negotiation so it is easy to set websocket stuff like extensions and subprotocol without having to actually set headers directly.